parquet/src/main/java/org/apache/iceberg/parquet/ParquetVariantUtil.java

```diff
@@ -395,7 +399,30 @@ public Type object(VariantObject object, List<String> names, List<Type> typedVal
 
     @Override
     public Type array(VariantArray array, List<Type> elementResults) {
-      return null;
+      if (elementResults.isEmpty()) {
+        return null;
```

Comment:
Does this not support shredding an array of encoded variants? This could be a list with only an inner `value` column?